packages/cmrjs/index.js

```diff
@@ -20,58 +20,37 @@ const logDetails = {
 };
 
 /**
- * Search for a concept on CMR
  *
- * @param {string} type - the concept type. Choices are: collection, granule
- * @param {Object} searchParams - cmr search params
+ * @param {string} type - Concept type to search, choices: ['collections', 'granules']
+ * @param {Object} searchParams - CMR search parameters
  * @param {Array} existingResults - array of results returned in previous recursive calls
- * @returns {Promise.<Array>} an array of search results
+ * @returns {Array} - array of search results.
  */
-async function searchConcept(type, searchParams, existingResults) {
-  const limit = process.env.CMR_LIMIT || 100;
-  const pageSize = process.env.CMR_PAGE_SIZE || 50;
+async function searchConcept(type, searchParams, existingResults = []) {
+  const limitRecords = process.env.CMR_LIMIT || 100;
+  const pageSize = searchParams.pageSize || process.env.CMR_PAGE_SIZE || 50;
   let pageNum = 1;
 
   if (searchParams.page_num) {
     pageNum = searchParams.page_num + 1;
   }
 
   // Recursively retrieve all the search results for collections or granules
-  // Also, parse them from XML into native JS objects
-  const qs = Object.assign(
-    Object.assign({ page_size: pageSize }, searchParams),
-    { page_num: pageNum }
-  );
-
-  const body = await got(getUrl('search') + type, { query: qs });
-
-  const str = await new Promise((resolve, reject) => {
-    parseString(body, xmlParseOptions, (err, res) => {
-      if (err) reject(err);
-
-      if (res.errors) {
-        const errorMessage = JSON.stringify(res.errors.error);
-        throw new Error(errorMessage);
-      }
-
-      resolve(res);
-    });
-  });
+  const queryObject = Object.assign({ page_size: pageSize }, searchParams, { page_num: pageNum });
```

Comment:
This might make it a little clearer what's going on, up to you.

```suggestion
  const defaultParams = { page_size: pageSize };
  cosnt queryObject = { ...defaultParams, ...searchParams, page_num: pageNum };
```